packages/shorebird_code_push_protocol/lib/src/models/build_environment_metadata.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'build_environment_metadata.g.dart';
+
+/// {@template build_environment_metadata}
+/// Information about the environment used to build a release or patch.
+/// {@endtemplate}
+@JsonSerializable()
+class BuildEnvironmentMetadata extends Equatable {
+  /// {@macro build_environment_metadata}
+  const BuildEnvironmentMetadata({
+    required this.shorebirdVersion,
+    required this.operatingSystem,
+    required this.operatingSystemVersion,
+    required this.xcodeVersion,
+  });
+
+  /// coverage:ignore-start
+  /// Creates a [BuildEnvironmentMetadata] with overridable default values for
+  /// testing purposes.
+  factory BuildEnvironmentMetadata.forTest({
+    String shorebirdVersion = '4.5.6',
+    String operatingSystem = 'macos',
+    String operatingSystemVersion = '1.2.3',
+    String? xcodeVersion = '15.0',
+  }) =>
+      BuildEnvironmentMetadata(
+        shorebirdVersion: shorebirdVersion,
+        operatingSystem: operatingSystem,
+        operatingSystemVersion: operatingSystemVersion,
+        xcodeVersion: xcodeVersion,
+      );
+  // coverage:ignore-end
+
+  /// Converts a Map<String, dynamic> to a [BuildEnvironmentMetadata]
+  factory BuildEnvironmentMetadata.fromJson(Map<String, dynamic> json) =>
+      _$BuildEnvironmentMetadataFromJson(json);
+
+  /// Converts a [BuildEnvironmentMetadata] to a Map<String, dynamic>
+  Map<String, dynamic> toJson() => _$BuildEnvironmentMetadataToJson(this);
+
+  /// The version of Shorebird used to run the command.
+  final String shorebirdVersion;
```

Comment:
Can you explain the reason next to each, e.g.:

```suggestion
  /// The version of Shorebird used to run the command.
  /// This is stored with the release and displayed in console.shorebird.dev.
  final String shorebirdVersion;
```

Reply:
Done